msdk-id-sirius/src/main/java/io/github/msdk/id/sirius/FingerIdWebMethod.java

```diff
@@ -0,0 +1,646 @@
+  private final static String FINGERID_SOURCE = "https://www.csi-fingerid.uni-jena.de";
+  /* TODO: This field must be altered whenever boecker-labs updates its API!!! */
+  private final static String FINGERID_VERSION = "1.1.2";
+  private final static SearchStructureByFormula searchDB = new RESTDatabase(BioFilter.ALL);
+  private final static Gson gson = new Gson();
+
+  private final Ms2Experiment experiment;
+  private final SiriusIonAnnotation ionAnnotation;
+  private final PredictionPerformance[] perf;
+  private final MaskedFingerprintVersion version;
+  private final Fingerblast blaster;
+  private final int candidatesAmount;
+  private List<IonAnnotation> newAnnotations;
+  private int finishedItems;
+  private boolean cancelled;
+
+
+  /**
+   * <p>Constructor for a FingerIdWebMethod</p>
+   * @param experiment - Ms2Experiment that was used to get ionAnnotation
+   * @param ionAnnotation - SiriusIonAnnotation returned from SiriusIdentificationMethod with FTree field specified
+   * @param candidatesAmount - amount of candidates to be returned from this method
+   * @throws MSDKException if any
+   */
+  public FingerIdWebMethod(Ms2Experiment experiment, SiriusIonAnnotation ionAnnotation, int candidatesAmount) throws MSDKException {
+    this.experiment = experiment;
+    this.ionAnnotation = ionAnnotation;
+
+    try {
+      final TIntArrayList list = new TIntArrayList(4096);
+      perf = getStatistics(getType(), list);
+      version = buildFingerprintVersion(list);
+      blaster = createBlaster(perf);
+    } catch (IOException e) {
+      throw new MSDKException(e);
+    }
+
+    newAnnotations = new LinkedList<>();
+    finishedItems = 0;
+    if (candidatesAmount == 0)
+      this.candidatesAmount = version.size();
+    else
+      this.candidatesAmount = candidatesAmount;
+  }
+
+  /**
+   * <p>Simplified FingerIdWebMethod constructor</p>
+   * In this case, FingerId will return unspecified amount of compounds (500-7000)
+   * @param experiment - Ms2Experiment object
+   * @param ionAnnotation - SiriusIonAnnotation with FTree specified
+   * @throws MSDKException if any
+   */
+  public FingerIdWebMethod(Ms2Experiment experiment, SiriusIonAnnotation ionAnnotation) throws MSDKException {
+    this(experiment, ionAnnotation, 0);
+  }
+
+  /**
+   * <p>Method returns FingerIdVersion</p>
+   * @return version
+   */
+  private static String fingerIdVersion() {
+    return FINGERID_VERSION;
+  }
+
+  /**
+   * <p>Method returns URIBuilder for the Web server</p>
+   * @param path API functionality you are interested in
+   * @return builder
+   * @throws URISyntaxException if any
+   */
+  private URIBuilder getFingerIdURI(String path) throws URISyntaxException {
+    if (path == null)
+      path = "";
+    URIBuilder builder = new URIBuilder(FINGERID_SOURCE);
+    builder.setPath("/csi-fingerid-" + FingerIdWebMethod.fingerIdVersion() + path);
```

Comment:
Why not use the constant FINGERID_VERSION instead of calling a private method fingerIdVersion()?

Reply:
Old idea, I used to think that it will be nice to add in future a capability of autochecking the version of FingerId
Removed.